packages/checkout/filter-registry/index.ts

```diff
@@ -211,42 +198,40 @@ export const __experimentalApplyCheckoutFilter = < T >( {
 	/** Function that needs to return true when the filtered value is passed in order for the filter to be applied. */
 	validation?: ( value: T ) => true | Error;
 } ): T => {
-	return useMemo( () => {
```

Comment:
I just tested this vs. trunk with some filters and they get called the same number of times. I think this is fine to remove.